src/utils/nanvixd/src/cache/mod.rs

```diff
@@ -181,4 +181,24 @@ impl SandboxCache{
 
         Ok(())
     }
+
+    ///
+    /// # Description
+    ///
+    /// Stop all instances in the cache.
+    ///
+    pub async fn cleanup(&mut self) -> Result<()> {
+        // TODO: for the time being only linuxd instances support being gracefully shutdown. User
+        // VMs are missing a control-plane socket.
+        for (tenant_id, linuxd_instance) in self.linuxd_instances.iter_mut() {
+            if let Some(linuxd_instance_mut) = Arc::get_mut(linuxd_instance) {
+                linuxd_instance_mut.shutdown().await?;
```

Comment:
We probably want to have thit in the drop method. We can address that in a later PR. If you agree could you create an issue for that?

Reply:
You suggest triggering the `cleanup` method inside the cache's drop?

I thought about this but the issue is that, given that cleanup involves sending network requests to other components, we need to `await` on them, and we cannot do that easily when implementing the `drop` trait (which is not async).